src/lib.rs

```diff
@@ -92,5 +92,25 @@ pub extern "C" fn __swift_bridge__null_pointer() -> *const std::ffi::c_void {
     std::ptr::null()
 }
 
+#[no_mangle]
```

Comment:
You can put this in under here https://github.com/chinedufn/swift-bridge/blob/23d60feaa0f8e92f2c854d8665868aa5573ef505/src/std_bridge/string.rs#L61-L82

Reply:
I got the link error like: 
```
Testing failed:
        Undefined symbol: ___swift_bridge__$RustStr$partial_eq
        Testing cancelled because the build failed.
```

How does the Swift-compiler know where methods are, such as `len`, `to_string`.
https://github.com/chinedufn/swift-bridge/blob/0de0dc880709ee655aa8d625d7bbefe420456b55/src/std_bridge/string.rs#L62-L80

Reply:
Seaching for `RustStr`'s methods with a tool like `grep`, I couldn't find any declarations of them in any c-header files.

Reply:
https://github.com/chinedufn/swift-bridge/blob/e8d3859d2a68b5d5e0012185a87e88b3c811f853/crates/swift-bridge-build/src/generate_core.rs#L16-L38

https://github.com/chinedufn/swift-bridge/blob/b654a24d6d5169e0e73192ebffa556cd43044512/crates/swift-bridge-build/src/generate_core/rust_string.c.h#L17-L18

Reply:
Oh wait, it looks like none of the `RustStr` methods like `len` or `to_string` are actually bridged.

But yeah you can declare the `___swift_bridge__$RustStr$partial_eq` inside of `rust_string.c.h`

Reply:
Also you can leave a module comment inside of `std_bridge/string.rs` along the lines of

```rust
//! The corresponding C and Swift code can be found in
//! crates/swift-bridge-build/src/generate_core/rust_string.{c.h,swift}
```

Same idea in `std_bridge/rust_vec.rs`

And then in `std_bridge.rs`

```rust
//! The corresponding C and Swift code can be found in
//! crates/swift-bridge-build/src/generate_core/*
```

Reply:
>But yeah you can declare the ___swift_bridge__$RustStr$partial_eq inside of rust_string.c.h 

I already tried to add `the ___swift_bridge__$RustStr$partial_eq` to `rust_string.c.h`. However I got the link error.

Reply:
https://github.com/NiwakaDev/swift-bridge/blob/03245ec39c4cf40f7909704456617f6186f1bc50/crates/swift-bridge-build/src/generate_core/rust_string.c.h#L19

Reply:
The build script might not be getting re-run.

Try running `touch crates/swift-integration-tests/build.rs` or otherwise modifying the `build.rs` so that it gets re-run and seeing if that works.

https://doc.rust-lang.org/cargo/reference/build-scripts.html#life-cycle-of-a-build-script

Reply:
>Try running touch crates/swift-integration-tests/build.rs or otherwise modifying and saving

I got the link error again. 

`RustString`'s methods seem to be imported on the Swift side.
<img width="446" alt="スクリーンショット 2023-01-30 1 33 35" src="https://user-images.githubusercontent.com/61189782/215340519-fe15894c-5ea8-4b29-b819-49b348b8a186.png">

However, `RustStr`'s methods doesn't seem to be imported on the Swift side.

<img width="583" alt="スクリーンショット 2023-01-30 1 33 44" src="https://user-images.githubusercontent.com/61189782/215340505-ab21bc84-dbb7-4a75-a676-b11b9b15cb37.png">

I tried to dive into this problem, and run `objdump -d libswift_integration_tests.a | grep RustStr`: 
```
0000000000000328 <__ZN12swift_bridge10std_bridge6string7RustStr10partial_eq17h39b2f23e5457666eE
```

`patial_eq` might be not `__swift_bridge__$RustStr$partial_eq` but `__ZN12swift_bridge10std_bridge6string7RustStr10partial_eq17h39b2f23e5457666eE`.

Reply:
I just cloned your branch and I can reproduce the error. Give me a sec

Reply:
This should work:

```rust
impl RustStr {
    pub fn len(&self) -> usize {
        self.len
    }

    // TODO: Think through these lifetimes and the implications of them...
    pub fn to_str<'a>(self) -> &'a str {
        let bytes = unsafe { std::slice::from_raw_parts(self.start, self.len) };
        std::str::from_utf8(bytes).expect("Failed to convert RustStr to &str")
    }

    pub fn to_string(self) -> String {
        self.to_str().to_string()
    }

    pub fn from_str(str: &str) -> Self {
        RustStr {
            start: str.as_ptr(),
            len: str.len(),
        }
    }
}

impl PartialEq for RustStr {
    fn eq(&self, other: &Self) -> bool {
        unsafe {
            std::slice::from_raw_parts(self.start, self.len)
                == std::slice::from_raw_parts(other.start, other.len)
        }
    }
}

#[export_name = "__swift_bridge__$RustStr$partial_eq"]
#[allow(non_snake_case)]
pub extern "C" fn __swift_bridge__RustStr_partial_eq(lhs: RustStr, rhs: RustStr) -> bool {
    lhs == rhs
}
```

Reply:
I'm sorry for taking up your time. I'm going to study rust hard.